test/executor_test.rb

```diff
@@ -146,7 +146,7 @@ def test_initialized_populates_index
   end
 
   def test_initialized_recovers_from_indexing_failures
-    RubyIndexer::Index.any_instance.expects(:index_all).once.raises(StandardError, "boom!")
```

Comment:
The idea of the `once` was to ensure that we weren't trying to index the codebase more than once in the `initialized` notification. I suppose the problem is that the thread switches and the stubbing is not thread safe.

Can we stub on the specific instance of the index instead? So that we continue to ensure it never gets indexed twice? Something like

```ruby
@executor.instance_variable_get(:@index).expects(:index_all).once.raises(StandardError, "boom!")
```

Reply:
Yeah we can do that, updated.